scripts/base/frameworks/analyzer/dpd.zeek

```diff
@@ -75,21 +82,37 @@ event analyzer_violation(c: connection, atype: AllAnalyzers::Tag, aid: count,
 	delete c$service[analyzer];
 	add c$service[fmt("-%s", analyzer)];
 
-	local info: Info;
-	info$ts=network_time();
-	info$uid=c$uid;
-	info$id=c$id;
-	info$proto=get_port_transport_proto(c$id$orig_p);
-	info$analyzer=analyzer;
-	info$failure_reason=reason;
-	c$dpd = info;
+	local dpd: Info;
+	dpd$ts = network_time();
+	dpd$uid = c$uid;
+	dpd$id = c$id;
+	dpd$proto = get_port_transport_proto(c$id$orig_p);
+	dpd$analyzer = analyzer;
+
+	# Encode data into the reason there's any as done for the old
+	# analyzer_violation event, previously.
+	local reason = info$reason;
+	if ( info?$data )
+		{
+		local ellipsis = |info$data| > 40 ? "..." : "";
+		local data = info$data[0:40];
+		reason = fmt("%s [%s%s]", reason, data, ellipsis);
+		}
+
+	dpd$failure_reason = reason;
+	c$dpd = dpd;
 	}
 
-event analyzer_violation(c: connection, atype: AllAnalyzers::Tag, aid: count, reason: string) &priority=5
+event analyzer_violation_info(atype: AllAnalyzers::Tag, info: AnalyzerViolationInfo ) &priority=5
 	{
 	if ( atype in ignore_violations )
 		return;
 
+	if ( ! info?$c || ! info?$aid )
```

Comment:
I'm wondering if we should indeed want to key this on having `c`? If we didn't, then a file analyzer could still set a connection as well. In your earlier version you had new bifs to check what type of analyzer it is, which I liked. How about going back to that?

Reply:
Same below for other tests for the type of analyzer.

Reply:
I've added the bifs back and wrappers respective modules and guarded the event handlers with those. Could also see them all in the global namespace as they become quite long to type and may fit there, too.

Reply:
Did you push those updates? Not seeing them. But yes, I can see them in the global namespace.

Reply:
Sorry about that. I just pushed that commit..

Reply:
They are now global as `is_protocol_analyzer()`, `is_packet_analyzer()`.